src/snowflake/cli/plugins/stage/manager.py

```diff
@@ -257,11 +262,14 @@ def _split_stage_path(self, stage_path: str) -> StagePathParts:
             stage_name -> stage
             directory -> dir
         """
+        directory = "/".join(Path(stage_path).parts[1:])
+
+        if self._is_user_stage(stage_path):
+            return StagePathParts(stage="@~", stage_name="@~", directory=directory)
```

Comment:
I wonder if there is a possibility to create a`Stage` object that would represent stage and has all the logic inside.
This way we could reference in code only to `Stage` object or its properties.

Reply:
That would be useful but probably out of scope for this Pr